Added dead mans switch contracts and base58 lib.

msinkec opened the pull request.

xhliu commented on contracts/deadMansSwitchNLocktime.scrypt:
```diff
@@ -0,0 +1,42 @@
+import "util.scrypt";
+
+/**
+ * Dead man's switch / last will contract.
```
Add comments explaining the difference btw the two versions of deadman

xhliu commented on contracts/deadMansSwitchStatefull.scrypt:
```diff
@@ -0,0 +1,66 @@
+    static const int MIN_BLOCK_TARGET = 128;
+    static const int MIN_N_BLOCKS_PASSED = 52500; // Around 1 year
+  
+    @state
+    PubKey pubKeyRefresh;
+  
+    @state
+    PubKey pubKeyCancel;
+  
+    @state
+    PubKey pubKeyInheritor;
+
+    @state
+    int inheritanceSats;
+
+    // The point of the refresh function is just to replace the old UTXO 
+    // with an exact copy in a newer block.
+    public function refresh(SigHashPreimage txPreimage, Sig sig) {
```
Add comments for each non-trivial property & function.

xhliu reviewed (approved):
🥇 looks good.


Add some comments before merging.

msinkec commented:
OK, I added some comments and as we discussed, I also removed the nLocktime version of dead mans switch, because we came to the conclusion, that the same goal can be achieved ussing a simple P2PKH using nLocktime.
